### PR TITLE
Add build-hpc workflow for Github Actions

### PR DESCRIPTION
**Pull request by reuterbal**
This builds with NVHPC on our GPU nodes in single and double precision, each with and without cuda.

### CODE REVIEW

**Review by awnawab — APPROVED** [review]
Many many thanks for this 🙏 Running the CI on GPU will be a big boost to the FIELD_API development process. G2G from me 👌

File: .github/workflows/build-hpc.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+jobs:
+  ci-hpc:
+    name: ci-hpc
+
+    strategy:
+      fail-fast: false    # false: try to complete all jobs
+
+      matrix:
+        prec: [dp, sp]
```
Comment (awnawab) [inline .github/workflows/build-hpc.yml:20]: By default FIELD_API will build both spdp libraries. Only a few files that are precision dependent are compiled twice, the rest are compiled only once. And all the tests are in fact run in DP, expect for one mixed precision test. So we don't need to extend the matrix to cover DP/SP builds. Sorry for missing this earlier.

**Review by awnawab — CHANGES_REQUESTED** [review]
One small nitpick about the testing matrix that I missed earlier, sorry!

**Comment by dareg** [comment]
Could you give to the workflow a name more similar to the other workflows and more explicit?
Something like FieldAPI_NVHPC_ECMWF_HPC.
Thanks for the PR.

**Comment by reuterbal** [comment]
In my opinion, the other workflows should be condensed into a single workflow with a configuration matrix - that is more in line with what we use in every other repository.

**Comment by awnawab** [comment]
> In my opinion, the other workflows should be condensed into a single workflow with a configuration matrix - that is more in line with what we use in every other repository.

PR #66 filed to do just that 😄

**Review by awnawab — APPROVED** [review]
Many thanks @reuterbal for addressing the comment, looks ready to merge to me now 👌

File: .github/workflows/build-hpc.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
+name: build-hpc
+
+# Controls when the action will run
+on:
+
+  # Trigger the workflow on all pull requests
```
Comment (mlange05) [inline .github/workflows/build-hpc.yml:6]: Don't we want to run this on `main` after merge-pushes like we usually do?

Comment (reuterbal) [reply .github/workflows/build-hpc.yml:13]: Thanks, good point! Fixed now

**Review by mlange05 — APPROVED** [review]
Looks great! GTG from me. :shipit:

**Review by dareg — APPROVED** [review]

**Pull request merged**